### PR TITLE
ci: test workflow

### LINKED ISSUE #19: What should be the minimum supported version of Node.js for this package?
I have doubts about what the minimum supported version of Node.js is for this package when setting the Node.js version in a workflow.

I assume it will be version 8 since it uses async/await, but it’s not clear.

### PR DESCRIPTION
**Pull request by bjohansebas**
A workflow is added to run tests in GitHub Actions, in addition to being able to know from which version this package is supported using the engines field in the package.json

note: More tests should be added since the package does not work correctly after version 14 of Node.js.

closes #19

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+  group: "test"
+  cancel-in-progress: false
+
+jobs:
+  test:
+
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        node-version: [10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20, 21, 22, 23]
```
Comment (wesleytodd) [inline .github/workflows/test.yml:17]: We can probably drop a lot of these older ones.

**Review by wesleytodd — APPROVED** [review]

File: .github/workflows/test.yml (hunk 1 above)
Comment (bjohansebas) [reply .github/workflows/test.yml:17]: From which version do you want to start testing?

Comment (bjohansebas) [reply .github/workflows/test.yml]: now use lts versions

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -0,0 +1,26 @@
+name: Run Tests
+
+on: 
+  push:
```
Comment (wesleytodd) [inline .github/workflows/test.yml:4]: Can we limit this to pushes to main?

Comment (bjohansebas) [reply .github/workflows/test.yml:4]: done

**Review by wesleytodd — APPROVED** [review]

**Pull request merged**